internal/kafka/internal/handlers/validation.go

```diff
@@ -145,21 +208,50 @@ func ValidateKafkaUserFacingUpdateFields(ctx context.Context, authService author
 	}
 }
 
-func ValidateKafkaClaims(ctx context.Context, kafkaRequestPayload *public.KafkaRequestPayload, kafkaRequest *dbapi.KafkaRequest) handlers.Validate {
-	return func() *errors.ServiceError {
-		kafkaRequest = presenters.ConvertKafkaRequest(*kafkaRequestPayload, kafkaRequest)
-		claims, err := auth.GetClaimsFromContext(ctx)
-		if err != nil {
-			return errors.Unauthenticated("user not authenticated")
+func getClaims(ctx context.Context) (jwt.MapClaims, *errors.ServiceError) {
+	claims, err := auth.GetClaimsFromContext(ctx)
+	if err != nil {
+		return nil, errors.Unauthenticated("user not authenticated")
+	}
+	return claims, nil
+}
+
+type ValidateKafkaClaimsOptions func(claims *jwt.MapClaims) *errors.ServiceError
+
+func ValidateUsername() ValidateKafkaClaimsOptions {
+	return func(claims *jwt.MapClaims) *errors.ServiceError {
+		username := auth.GetUsernameFromClaims(*claims)
+		if username == "" {
+			return errors.New(errors.ErrorBadRequest, "can't find 'user' attribute in claims")
```

Comment:
`user` in the error reason message can be misleading. The claims we are interested are configurable at https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/4e3302d57225475aad5e54e286041946fe8a4c8b/pkg/auth/context.go#L25 or the alternative one at https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/4e3302d57225475aad5e54e286041946fe8a4c8b/pkg/auth/context.go#L30 so I think we either:
- include the right claim key in the message 
- or reword the message a bit to make it generic and avoid sending the claim key which can be confusing to the consumer of the API